hooks/configure

```diff
@@ -14,74 +14,108 @@
+    with open(conf+".tmp", "w") as fp:
+        fp.write("[Login]\nHandlePowerKey={}\n".format(action))
+    os.rename(conf+".tmp", conf)
+
+
+def switch_service(service, enable):
+    srv = "{}.service".format(service)
+    if enable == "true":
+ 	# A simple `systemctl disable --now ssh.service` doesn't work
+ 	# and fails with an error message. Because of that we're going
+ 	# in two steps here and disable first and then stopping the
+ 	# service unit.
+        systemctl("disable", srv)
+        systemctl("stop", srv)
+    elif enable == "false":
+ 	# When the unit is already enabled but not active a call with
+ 	# --now does start the unit so we have to check for that  case
+ 	# and explicitly start the unit.
+        if systemctl("is-enabled", srv).stdout.strip() == "disabled":
+            systemctl("enable", srv, check=True)
+        if systemctl("is-active", srv).stdout.strip() == "inactive":
+            systemctl("start", srv, check=True)
+    else:
+        raise Exception("Invalid value '{}' provided for option {}".format(enable, srv))
+
+
+def update_pi_config_value(path, key, value):
+    with open(path) as r, open(path+".tmp", "w") as w:
+        found=False
+        for line in r:
+            m=re.match(r'^[ \t#]*({}=.*)$'.format(key), line)
+            if m:
+                if value == "":
+                    line="#{}\n".format(m.group(1))
+                else:
+                    line="{}={}\n".format(key, value)
+                found=True
+            w.write(line)
+        if not found and value != "":
+            w.write("{}={}\n".format(key, value))
+    os.rename(path+".tmp", path)
+            
+
+if __name__ == "__main__":
+    if not core_support_available():
+        print("Cannot run systemctl - is core-support available?")
+        sys.exit(0)
+
+    for service in SERVICES:
+        value=check_output(["snapctl", "get", "service.{}.disable".format(service)], universal_newlines=True).strip()
+        if value != "":
+            switch_service(service, value)
+    
+    powerkey=check_output(["snapctl", "get", "system.power-key-action"], universal_newlines=True).strip()
+    if powerkey != "":
+        switch_handle_power_key(powerkey)
+
+    PI_CONFIG_KEYS=[
+        "disable_overscan", "framebuffer_width", "framebuffer_height",
+        "framebuffer_depth", "framebuffer_ignore_alpha", "overscan_left",
+        "overscan_right", "overscan_top", "overscan_bottom", "overscan_scale",
+        "display_rotate", "hdmi_group", "hdmi_mode", "hdmi_drive",
+        "avoid_warnings", "gpu_mem_256", "gpu_mem_512", "gpu_mem",
+    ]
+    PI_CONFIG=os.getenv("TEST_UBOOT_CONFIG", default="/boot/uboot/config.txt")
+    if os.path.isfile(PI_CONFIG):
+        for key in PI_CONFIG_KEYS:
+            value=check_output(["snapctl", "get", "pi-config.{}".format(key.replace("_","-"))], universal_newlines=True).strip()
+            update_pi_config_value(PI_CONFIG, key, value)
```

Comment:
we're writing out the same file multiple times every time, to change a single value in each pass? Am I reading this right?
And we're not syncing any of those writes
This is Not Good